internal/summon/command/run.go

```diff
@@ -37,20 +38,48 @@ func buildEnvironment(secrets map[string]string, secretsMap secretsyml.SecretsMa
 // resolveSecrets obtains the value of each requested secret.
 func resolveSecrets(provider plugin_v1.Provider, secretsMap secretsyml.SecretsMap) (result map[string]string, err error) {
 	result = make(map[string]string)
+
+	var varSecretsSpecKeys []string
+	var varSecretsSpecPaths []string
+
 	for key, spec := range secretsMap {
-		var value string
 		if spec.IsVar() {
-			var valueBytes []byte
-			if valueBytes, err = provider.GetValue(spec.Path); err != nil {
-				return
-			}
-			value = string(valueBytes)
+			varSecretsSpecKeys = append(varSecretsSpecKeys, key)
+			varSecretsSpecPaths = append(varSecretsSpecPaths, spec.Path)
 		} else {
 			// If the spec isn't a variable, use its value as-is
-			value = spec.Path
+			value := spec.Path
+			result[key] = value
+		}
+	}
+
+
+	// Get the variable values
+	providerResponses, err := provider.GetValues(varSecretsSpecPaths...)
+	if err != nil {
+		return
```

Comment:
Not a blocker but the use of function def line variables makes it really difficult to tell that you're not returning a success here

Reply:
Addressed